src/pwm.rs

```diff
@@ -44,41 +44,48 @@ macro_rules! pins_to_channels_mapping {
 
 pins_to_channels_mapping! {
     // TIM1
-    TIM1: (PA8, PA9, PA10, PA11), (C1, C2, C3, C4);
-    TIM1: (PA9, PA10, PA11), (C2, C3, C4);
-    TIM1: (PA8, PA10, PA11), (C1, C3, C4);
-    TIM1: (PA8, PA9, PA11), (C1, C2, C4);
-    TIM1: (PA8, PA9, PA10), (C1, C2, C3);
-    TIM1: (PA10, PA11), (C3, C4);
-    TIM1: (PA9, PA11), (C2, C4);
-    TIM1: (PA9, PA10), (C2, C3);
-    TIM1: (PA8, PA11), (C1, C4);
-    TIM1: (PA8, PA10), (C1, C3);
-    TIM1: (PA8, PA9), (C1, C2);
-    TIM1: (PA8), (C1);
-    TIM1: (PA9), (C2);
-    TIM1: (PA10), (C3);
-    TIM1: (PA11), (C4);
+    TIM1: (PA8, PA9, PA10, PA11), (C1, C2, C3, C4), (AF1, AF1, AF1, AF1);
+    TIM1: (PA9, PA10, PA11), (C2, C3, C4), (AF1, AF1, AF1);
+    TIM1: (PA8, PA10, PA11), (C1, C3, C4), (AF1, AF1, AF1);
+    TIM1: (PA8, PA9, PA11), (C1, C2, C4), (AF1, AF1, AF1);
+    TIM1: (PA8, PA9, PA10), (C1, C2, C3), (AF1, AF1, AF1);
+    TIM1: (PA10, PA11), (C3, C4), (AF1, AF1);
+    TIM1: (PA9, PA11), (C2, C4), (AF1, AF1);
+    TIM1: (PA9, PA10), (C2, C3), (AF1, AF1);
+    TIM1: (PA8, PA11), (C1, C4), (AF1, AF1);
+    TIM1: (PA8, PA10), (C1, C3), (AF1, AF1);
+    TIM1: (PA8, PA9), (C1, C2), (AF1, AF1);
+    TIM1: (PA8), (C1), (AF1);
+    TIM1: (PA9), (C2), (AF1);
+    TIM1: (PA10), (C3), (AF1);
+    TIM1: (PA11), (C4), (AF1);
 
     // TIM2
-    TIM2: (PA0, PA1, PA2, PA3), (C1, C2, C3, C4);
-    TIM2: (PA1, PA2, PA3), (C2, C3, C4);
-    TIM2: (PA0, PA2, PA3), (C1, C3, C4);
-    TIM2: (PA0, PA1, PA3), (C1, C2, C4);
-    TIM2: (PA0, PA1, PA2), (C1, C2, C3);
-    TIM2: (PA2, PA3), (C3, C4);
-    TIM2: (PA1, PA3), (C2, C4);
-    TIM2: (PA1, PA2), (C2, C3);
-    TIM2: (PA0, PA3), (C1, C4);
-    TIM2: (PA0, PA2), (C1, C3);
-    TIM2: (PA0, PA1), (C1, C2);
-    TIM2: (PA0), (C1);
-    TIM2: (PA1), (C2);
-    TIM2: (PA2), (C3);
-    TIM2: (PA3), (C4);
+    TIM2: (PA0, PA1, PA2, PA3), (C1, C2, C3, C4), (AF1, AF1, AF1, AF1);
+    TIM2: (PA1, PA2, PA3), (C2, C3, C4), (AF1, AF1, AF1);
+    TIM2: (PA0, PA2, PA3), (C1, C3, C4), (AF1, AF1, AF1);
+    TIM2: (PA0, PA1, PA3), (C1, C2, C4), (AF1, AF1, AF1);
+    TIM2: (PA0, PA1, PA2), (C1, C2, C3), (AF1, AF1, AF1);
+    TIM2: (PA2, PA3), (C3, C4), (AF1, AF1);
+    TIM2: (PA1, PA3), (C2, C4), (AF1, AF1);
+    TIM2: (PA1, PA2), (C2, C3), (AF1, AF1);
+    TIM2: (PA0, PA3), (C1, C4), (AF1, AF1);
+    TIM2: (PA0, PA2), (C1, C3), (AF1, AF1);
+    TIM2: (PA0, PA1), (C1, C2), (AF1, AF1);
+    TIM2: (PA0), (C1), (AF1);
+    TIM2: (PA1), (C2), (AF1);
+    TIM2: (PA2), (C3), (AF1);
+    TIM2: (PA3), (C4), (AF1);
 
     // TIM15
-    TIM15: (PB14), (C1);
+    TIM15: (PB14), (C1), (AF14);
+    // TIM15: (PB15), (C2), (AF14);
+    TIM15: (PA2), (C1), (AF14);
+    // TIM15: (PA3), (C2), (AF14);
```

Comment:
Why are these commented out? Are these the possible configurations for all chips? In which case could you put in a TODO comment for feature gating them in the future.

Reply:
The commented lines are due to incompatibility issues with the PAC that does not have all the registers of the MCU yet, so once the PAC is updated with support these can be un-commented and all will start to work.